lib/python/qmk/commands.py

```diff
@@ -100,23 +106,43 @@ def get_git_version(repo_dir='.', check_dir='.'):
         else:
             cli.log.warn(f'"{" ".join(git_describe_cmd)}" returned error code {git_describe.returncode}')
             print(git_describe.stderr)
-            return strftime(time_fmt)
+            return current_time
 
-    return strftime(time_fmt)
+    return current_time
 
 
-def write_version_h(git_version, build_date, chibios_version, chibios_contrib_version):
+def write_version_h(skip_git=False, skip_all=False):
     """Generate and write quantum/version.h
     """
-    version_h = [
+    if skip_all:
+        current_time = "2020-01-01-00:00:00"
+    else:
+        current_time = strftime(time_fmt)
+
+    if skip_git:
+        git_version = "NA"
+        chibios_version = "NA"
+        chibios_contrib_version = "NA"
+    else:
+        git_version = get_git_version(current_time)
+        chibios_version = get_git_version(current_time, "chibios", "os")
+        chibios_contrib_version = get_git_version(current_time, "chibios-contrib", "os")
+
+    version_h_lines = [
+        '/* This file was automatically generated. Do not edit or copy.',
+        ' */',
+        '',
+        '#pragma once',
+        '',
         f'#define QMK_VERSION "{git_version}"',
-        f'#define QMK_BUILDDATE "{build_date}"',
+        f'#define QMK_BUILDDATE "{current_time}"',
         f'#define CHIBIOS_VERSION "{chibios_version}"',
         f'#define CHIBIOS_CONTRIB_VERSION "{chibios_contrib_version}"',
+        '',
     ]
```

Comment:
This is probably a bit easier to read/extend with an f-stringified heredoc:

```python
    version_h_text = f"""/* This file was automatically generated by `qmk generate-version-h`. Do not edit or copy.
 */

#pragma once

#define QMK_VERSION "{git_version}"
#define QMK_BUILDDATE "{current_time}"
#define CHIBIOS_VERSION "{chibios_version}"
#define CHIBIOS_CONTRIB_VERSION "{chibios_contrib_version}"

"""

Reply:
Just following how it's done for other generated files:
https://github.com/qmk/qmk_firmware/blob/master/lib/python/qmk/cli/generate/config_h.py#L99

Reply:
I think `generate/config_h` is a bit different due to conditionals and it could also be a bit simplified.
We are not too consistent with these generators, see [`generate/rgb_breathe_table](https://github.com/qmk/qmk_firmware/blob/master/lib/python/qmk/cli/generate/rgb_breathe_table.py#L54).

In any case, the current code works, so I'm not against merging it as is, just don't see much value in the list-based approach here.

Reply:
The original pattern is a habit I developed because in most situations it's faster to build an array of strings and concatenate at the end than to continually extend a string. For our situation I don't think the performance difference matters enough.